tttClassUML.puml

```diff
@@ -0,0 +1,48 @@
+'visit the link to for instructions on how to install GraphViz in order to generate graph within intelliJ https://plantuml.com/graphviz-dot
+@startuml
+
+class Player {
+    -String marker
+    -boolean isComputer
+
+    +Player()
+    +String : getMarker()
+    +boolean : isComputerPlayer()
+}
+
+class Board {
+    -Array[] board
+
+    +Board()
+    +Array[]: getBoard()
+    +String : getSpace(location : int)
+    +void : updateBoard()
+}
+
+class Game {
+    +String currentPlayer
+    -Validation validate
+    -Board board
+    -Player player1
+    -Player player2
+
+    +Game()
+    +void : initializePlayers()
+    +void : initializeGame()
+    +void : nextPlayer()
+    +void : move(currentPlayer)
+    -void : gameloop()
+}
+
+class Validation {
+    -Array[][] winningMoves
+
+    +boolean : someoneHasWon()
+    +boolean : gameIsTied()
+    +boolean : validMove()
```

Comment:
To me this feels like a question of "how smart should the Board be?"

The validMove seems related to the Board, but I might argue that the Game is responsible for the other game ending validation methods

Reply:
I would say anything in the prior TTT game should not be automatically be assumed to exist in the new one. The abstractions from the last one were in response to that codebase to help learn individual concepts, not necessarily the best design choices themselves. The primary goal of this re-do is to allow us to step back from that exercise.

For the concept of a validation class, I would consider one when there are validations at a service/application level. Where it would have validation logic that spans relationships of multiple parent objects working together. -- Starting with validations inside of models to confirm that they are valid is a very reasonable thing to do. An object is told to perform an action and it uses its internal data to make sure that is OK and then continues with the processing is a really good thing to drive out stronger understandings.

Reply:
It was definitely a valid suggestion from @billwanjohi that took some time for me to get my head around it. The main point of contention I had for keeping `validMoves` abstracted was that it needed to know the markers the players were using while it checked if the space was available. The actual statement should've been: the board should only update if there are no markers on the target location." Would everyone agree for the meantime the Validation class should be scrapped or should it still exist as a utility class for game validation?

Reply:
My direct answer would be, no you do not need it. Maybe further in your design, it will drive out the need for it, likely not though.

I would also challenge the assumption that a "validMoves" check would need knowledge of what a marker is.
To have a player take ownership over a cell on the board, what would the board know about or care about to make that happen? Would it be the display marker, would it be a player, an ownership id, something else? Whichever you pick, when that concept changes for its own reasons, the board also has to change. Which is more stable, which makes most sense for what the object/class is and is responsible for?